src/domains/Auth.ts

```diff
@@ -0,0 +1,73 @@
+import { getCookie } from '../utils';
+import { AUTH_BASE_URL } from '../constants';
+
+export async function signup(email: string, name: string, password: string) {
+  const response = await fetch(`${AUTH_BASE_URL}/register`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      email,
+      name,
+      password,
+    }),
+  });
+  const jsonResponse = await response.json();
+
+  if (!jsonResponse.accessToken) {
+    throw new Error(jsonResponse);
+  }
+}
+
+export async function login(email: string, password: string) {
+  const response = await fetch(`${AUTH_BASE_URL}/login`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      email,
+      password,
+    }),
+  });
+  const jsonResponse = await response.json();
+
+  if (!jsonResponse.accessToken) {
+    throw new Error(jsonResponse);
+  }
+
+  document.cookie = `user_id=${jsonResponse.user.id}`;
+  document.cookie = `access_token=${jsonResponse.accessToken}`;
+}
+
+export function logout() {
+  document.cookie = `user_id=`;
+  document.cookie = `access_token=`;
+}
+
+export function isLoggedIn() {
+  const userId: string = getCookie('user_id');
+  const accessToken: string = getCookie('access_token');
+
+  return userId && userId !== '' && accessToken && accessToken !== '';
+}
+
+export async function getUser(): Promise<object> {
+  const userId: string = getCookie('user_id');
+  const accessToken: string = getCookie('access_token');
+
+  if (!isLoggedIn()) return undefined;
+
+  const response = await fetch(`${AUTH_BASE_URL}/users/${userId}`, {
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${accessToken}`,
+    },
+  });
+
+  const user: Promise<object> = await response.json();
+
+  return user;
```

Comment:
getUser 함수가 어떤 타입을 반환할지 명시하고 있어서 따로 변수를 거치지 않고 return 해도 될 것 같으네요!
```suggestion
  return response.json();
```

Reply:
동의합니다 선언한 변수를 가지고 뭔가를 더 하는 게 아니니 바로 return 하는 게 더 좋겠네요! [d0dc24c](https://github.com/woowacourse/javascript-vendingmachine/pull/71/commits/d0dc24c324de6ac7b7a56b6485449d33bf8fe579)